packages/core/modules/requester/api-key.js

```diff
@@ -9,27 +10,42 @@ class ApiKeyRequester extends Requester {
     constructor(params) {
         super(params);
         this.requesterType = 'apiKey';
-        this.API_KEY_NAME = 'key';
-        this.API_KEY_VALUE = null;
+
+        // Use snake_case convention consistent with OAuth2Requester and BasicAuthRequester
+        this.api_key_name = get(params, 'api_key_name', 'key');
+        this.api_key = get(params, 'api_key', null);
+
+        // Backward compatibility: support old naming convention
+        if (!this.api_key && params.API_KEY_VALUE) {
```

Comment:
do we have to maintain backward compatibility?

Reply:
Currently I'm not sure what state api key management/persistence is in, so this felt safe.